pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -450,7 +450,7 @@ func (s *ServiceEntryStore) GetService(hostname host.Name) (*model.Service, erro
 }
 
 func (s *ServiceEntryStore) getServices() []*model.Service {
-	services := make([]*model.Service, 0)
+	services := make([]*model.Service, 0, len(s.store.ServiceEntries()))
```

Comment:
please don't do this! `s.store.ServiceEntries()` is not free

Reply:
Valid point. convertServices... will cause size issue. Done